File: clients/go/examples/balance/balance.go
package main

import (
	"context"
	"errors"
	"fmt"
	"io"

	"github.com/ethereum/go-ethereum/common"
	"github.com/ethereum/go-ethereum/ethclient"
	sidecar "github.com/zama-ai/sdk/clients/go"
)

func showBalance(ctx context.Context, provider *ethclient.Client, sdk *sidecar.SDKContext, token, owner common.Address, output io.Writer) error {
	name, encrypted, err := readToken(ctx, provider, token, owner)
	if err != nil {
		return err
	}
	if _, err := fmt.Fprintf(output, "User Address: %s\nToken: %s (%s) https://eth-sepolia.blockscout.com/token/%s\nEncrypted balance: %s\n", owner.Hex(), name, token.Hex(), token.Hex(), encrypted.Hex()); err != nil {
		return err
	}
	values, err := sdk.DecryptValues(ctx, []sidecar.EncryptedInput{{EncryptedValue: encrypted, ContractAddress: token}}, sidecar.DecryptOptions{})
	if err != nil {
		return err
	}
	balance, ok := values[encrypted]
	if !ok || balance.Kind != sidecar.ClearBigInt {
		return errors.New("decryption returned no integer balance")
	}
	_, err = fmt.Fprintf(output, "Decrypted balance: %s\n", balance.Integer)
	return err
}

File: clients/rust/examples/balance/balance.rs
use alloy_provider::Provider;
use alloy_sol_types::sol;
use anyhow::Result;
use zama_sdk::{Address, ClearValue, EncryptedInput, Sdk};

sol! {
    #[sol(rpc)]
    interface ConfidentialToken {
        function name() external view returns (string);
        function confidentialBalanceOf(address account) external view returns (bytes32);
    }
}

pub async fn show_balance(
    sdk: &Sdk,
    provider: &impl Provider,
    token: Address,
    owner: Address,
) -> Result<()> {
    let ctoken = ConfidentialToken::new(token, provider);
    let name = ctoken.name().call().await?;
    let encrypted = ctoken.confidentialBalanceOf(owner).call().await?;
    let address = token.to_checksum(None);
    println!("User Address: {}", owner.to_checksum(None));
    println!("Token: {name} ({address}) https://eth-sepolia.blockscout.com/token/{address}");
    println!("Encrypted balance: {encrypted:#x}");
    let values = sdk
        .decryption()
        .decrypt_values(
            &[EncryptedInput {
                encrypted_value: encrypted,
                contract_address: token,
            }],
            None,
        )
        .await?;
    match values.get(&encrypted) {
        Some(ClearValue::BigInt(balance)) => println!("Decrypted balance: {balance}"),
        _ => anyhow::bail!("unexpected decrypted balance type"),
    }
    Ok(())
}
